packages/go_router/lib/src/delegate.dart

```diff
@@ -59,13 +61,12 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       final RouteBase route = match.route;
 
       if (route is GoRoute && route.parentNavigatorKey != null) {
-        // It should not be possible for a GoRoute with parentNavigatorKey to be
-        // the only page, so maybePop should never return false in this case.
-        assert(await route.parentNavigatorKey!.currentState!.maybePop());
-        return true;
+        return route.parentNavigatorKey!.currentState!.maybePop();
```

Comment:
we may need to worry about cases like this

[ShellRoute(A), GoRoute(null), ShellRoute(B) GoRoute(B)]

in this case the GoRoute B will try to pop ShellRouteB and if it return false, we need to keep search up to ShellRoute A

Reply:
I had to remove the assertion from the ShellRoute constructor to do this... Are you sure that's what you want? 9a514615cb54882472ecbf9eebf48295c1716788

Reply:
I think we can allow it by improving the assertion: [17d90c4](https://github.com/flutter/packages/pull/2652/commits/17d90c44453e328851616b01593bd8f2bb8c600a)